internal/controller/example_extension_reconciler.go

```diff
@@ -0,0 +1,33 @@
+package controllers
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"k8s.io/client-go/dynamic"
+	ctrl "sigs.k8s.io/controller-runtime"
+	ctrlruntimemanager "sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var _ reconcile.Reconciler = &ExampleExtensionReconciler{}
+
+type ExampleExtensionReconciler struct {
+	client *dynamic.DynamicClient
+	logger logr.Logger
+}
+
+func NewExampleReconciler(client *dynamic.DynamicClient, logger logr.Logger) *ExampleExtensionReconciler {
+	return &ExampleExtensionReconciler{
+		client: client,
+		logger: logger,
+	}
+}
+
+func (e *ExampleExtensionReconciler) Reconcile(_ context.Context, _ reconcile.Request) (reconcile.Result, error) {
```

Comment:
Right now this is a standard controller, but we need to create a wrapper that provides the KuadrantCtx that this would implement